src/components/ReportsPage/HeatmapPage/SingleScorecardHeatmap.tsx

```diff
@@ -13,38 +13,74 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { useMemo } from 'react';
 import { Progress, WarningPanel } from '@backstage/core-components';
-import { useCortexApi, useEntitiesByTag } from '../../../utils/hooks';
-import { GroupByOption, HeaderType } from '../../../api/types';
+import { useCortexApi } from '../../../utils/hooks';
 import { SingleScorecardHeatmapTable } from './Tables/SingleScorecardHeatmapTable';
+import { StringIndexable, applyScoreFilters, catalogToRelationsByEntityId } from './HeatmapUtils';
+import { HomepageEntity } from '../../../api/userInsightTypes';
+import { HeatmapPageFilters } from './HeatmapFilters';
+import { ScorecardLadder } from '../../../api/types';
+import { keyBy } from 'lodash';
 
 interface SingleScorecardHeatmapProps {
   entityCategory: string;
+  entitiesByTag: StringIndexable<HomepageEntity>;
   scorecardId: number;
-  groupBy: GroupByOption;
-  headerType: HeaderType;
+  ladder: ScorecardLadder | undefined;
+  filters: HeatmapPageFilters;
 }
 
 export const SingleScorecardHeatmap = ({
   entityCategory,
+  entitiesByTag,
   scorecardId,
-  groupBy,
-  headerType,
+  ladder,
+  filters: {
+    groupBy,
+    headerType,
+    scoreFilters,
+    useHierarchy,
+    hideWithoutChildren,
+  }
 }: SingleScorecardHeatmapProps) => {
   const {
     value: scores,
     loading: loadingScores,
     error: scoresError,
   } = useCortexApi(api => api.getScorecardScores(scorecardId), [scorecardId]);
 
-  const { value: ladders, loading: loadingLadders } = useCortexApi(
-    api => api.getScorecardLadders(scorecardId),
-    [scorecardId],
-  );
-  const { entitiesByTag, loading: loadingEntities } = useEntitiesByTag();
+  const { ownerEmailByEntityId, groupTagByEntityId } = useMemo(() => {
+    return catalogToRelationsByEntityId(entitiesByTag);
+  }, [entitiesByTag]);
+
+  const { value: domainIdByEntityId, error: domainByEntityError, loading: isLoadingDomainByEntity } = useCortexApi(api => api.getEntityDomainAncestors());
+  const domainTagByEntityId = useMemo(() => {
+    if (!domainIdByEntityId) {
+      return {};
+    }
+
+    const map = {} as Record<string, string[]>;
+    const domainsById = keyBy(Object.values(entitiesByTag).filter((entity) => entity.type === "domain"), "id");
```

Comment:
nit: prefer function syntax for type safety
```suggestion
    const domainsById = keyBy(Object.values(entitiesByTag).filter((entity) => entity.type === "domain"), (domain) => domain.id);
```